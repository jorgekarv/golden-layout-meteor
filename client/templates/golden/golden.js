Template.golden.onRendered(function () {

  var config = {
    content: [{
        type: 'row',
        content:[{
            type: 'component',
            componentName: 'testComponent',
            componentState: { id: 'id_1', label: 'a' }
        },{
            type: 'column',
            content:[{
                type: 'component',
                componentName: 'testComponent',
                componentState: { id: 'id_2', label: 'b' }
            },{
                type: 'component',
                componentName: 'testComponent',
                componentState: { id: 'id_3', label: 'c' }
            }]
        }]
    }]
  };

  this.myLayout = new GoldenLayout( config );

  this.myLayout.registerComponent( 'testComponent', function( container, componentState ){
    container.getElement().html( '<div class="'+ componentState.id +'">' + componentState.label + '</div>' );
  });

  this.myLayout.init();

  var data = Animals.find({},{ sort: {rank: 1}});
  var templateHTML = Blaze.renderWithData(Template.goldenList,data,$('.id_1')[0]);

});


Template.golden.onDestroyed(function () {

     this.myLayout.destroy();

});
