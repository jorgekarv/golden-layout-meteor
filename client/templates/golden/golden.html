<template name="golden">

    <p>Golden layout tabs</p>
    <div id="layoutContainer"></div>

</template>
